app/src/main/java/com/sunasterisk/youtubebackground/data/source/VideoRepository.kt

```diff
@@ -0,0 +1,21 @@
+package com.sunasterisk.youtubebackground.data.source
+
+import com.sunasterisk.youtubebackground.data.source.local.VideoLocalDataSource
+import com.sunasterisk.youtubebackground.data.source.remote.VideoRemoteDataSource
+
+class VideoRepository private constructor(
+    private val localDataSource: VideoDataSource.Local?,
+    private val remoteDataSource: VideoDataSource.Remote?
+){
```

Comment:
data source không thể null 
```suggestion
class VideoRepository private constructor(
    private val localDataSource: VideoDataSource.Local,
    private val remoteDataSource: VideoDataSource.Remote
){
```